src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1340,6 +1340,10 @@ private static bool IsLessDerivedThanAny<TMember>(int index, TypeSymbol type, Ar
                 {
                     return true;
                 }
+                else if (currentType.IsExtension && currentType.ExtendedTypeIsOrDerivedFrom(type, TypeCompareKind.ConsiderEverything, useSiteInfo: ref useSiteInfo))
```

Comment:
>else if (currentType.IsExtension && currentType.ExtendedTypeIsOrDerivedFrom(type, TypeCompareKind.ConsiderEverything, useSiteInfo: ref useSiteInfo)) [](http://example.com/codeflow?start=16&length=148)

It doesn't look like this code is going to handle scenarios when extended type is an interface.